My_work_3/Simple_task_2.cpp

```diff
@@ -0,0 +1,45 @@
+#include <iostream>
+#include <cmath>
+#include <vector>
+#include <fstream>
+
+int main()
+{
+    int T;
+    double dt, k, R;
+
+    T = 100;
+    R = 3400000;
+    dt = 0.1;
+    k = 1.0d;
+
+    int iter = T/dt;
+//    std::vector<double> a(iter, 0);
+//    std::vector<double> b(iter, 0);
+//    std::vector<double> time(iter, 0);
+//    a[0] = R;
+//    for(int i=0; i<iter-1; ++i){
+//        a[i+1] = a[i] + b[i]*dt;
+//        b[i+1] = b[i] - a[i]*k*dt;
+//        time[i+1] = time[i] + dt;
+//    }
+    double a_old = R;
+    double b_old = 0.0;
+    double a_new, b_new, t = 0.0;
+
+    std::ofstream out;
+    out.open("Data.txt");
+    if (out.is_open()){
+        for(int i=0; i<iter; ++i){
+            a_new = a_old + b_old*dt;
+            b_new = b_old - a_old*k*dt;
+            out << t << ';' << a_old << ';' << b_old << '\n';
```

Comment:
How much memory does your program consume? How much storage? What will happen in case you are willing to store 1e12 timesteps results? How to solve this issue? 

Generally it's a common sense to store only fraction of the total simulation data. Say 1 of 100 or 1 of N of states. This approach is crucial in case you are willing to store all of the steps in RAM.

Btw streaming the results to file would hinder the code in case you are trying to compare performance of the cpp and python codes

Reply:
In common practice it is almost always a performance-memory tradeof. It's too time-consuming to write each time step separately (due to slow Input/Output disk operations in comparison with RAM access times) and too memory consuming to store every timestep data in RAM. So usually there is a buffer for the data from a few previous iterations to be saved, and daving process works only once in a few iterations or once the buffer is full. In that case you can use reasonably large buffers and make I/O operations relatively fast in a bufferized way. My comment was to highlight the problems with cpp buffered streaming input/output operations once it is done on every iteration.